biojava-core/src/main/java/org/biojava/nbio/core/util/FileDownloadUtils.java

```diff
@@ -161,6 +164,104 @@ public static void downloadFile(URL url, File destination) throws IOException {
 		tempFile.delete();
 
 	}
+	
+	/**
+	 * Creates validation files beside a file to be downloaded.<br>
+	 * Whenever possible, for a <code>file.ext</code> file, it creates 
+	 * <code>file.ext.size</code> and <code>file.hash</code> for in the same 
+	 * folder where <code>file.ext</code> exists.
+	 * If the file connection size could not be deduced from the URL, no size file is created. 
+	 * If <code>hashURL</code> is <code>null</code>, no hash file is created.
+	 * @param url the remote file URL to download
+	 * @param localDestination the local file to download into
+	 * @param hashURL the URL of the hash file to download. Can be <code>null</code>.
+	 */
+	public static void createValidationFiles(URL url, File localDestination, URL hashURL){
+		try {
+			URLConnection resourceConnection = url.openConnection();
+			createValidationFiles(resourceConnection, localDestination, hashURL);
+		} catch (IOException e) {
+			logger.warn("could not open connection to resource file due to exception", e);
+		}
+	}
+	/**
+	 * Creates validation files beside a file to be downloaded.<br>
+	 * Whenever possible, for a <code>file.ext</code> file, it creates 
+	 * <code>file.ext.size</code> and <code>file.hash</code> for in the same 
+	 * folder where <code>file.ext</code> exists.
+	 * If the file connection size could not be deduced from the resourceUrlConnection 
+	 * {@link URLConnection}, no size file is created. 
+	 * If <code>hashURL</code> is <code>null</code>, no hash file is created.
+	 * @param resourceUrlConnection the remote file URLConnection to download
+	 * @param localDestination the local file to download into
+	 * @param hashURL the URL of the hash file to download. Can be <code>null</code>.
+	 * @since 6.0.6
+	 */
+	public static void createValidationFiles(URLConnection resourceUrlConnection, File localDestination, URL hashURL){
+		long size = resourceUrlConnection.getContentLengthLong();
+		if(size == -1) {
+			logger.warn("could not find expected file size for resource {}.", resourceUrlConnection.getURL());
+		} else {
+			System.out.println("Content-Length: " + size);
```

Comment:
This should be done with logger. Probably at debug level

Reply:
Done.